src/main/java/com/redhat/devtools/intellij/quarkus/module/QuarkusCodeEndpointChooserStep.java

```diff
@@ -115,34 +147,117 @@ private void updateCustomUrl() {
         this.endpointURL.getTextEditor().setEditable(custom);
         this.endpointURL.setEnabled(custom);
         this.customUrlWithBrowseButton.setButtonEnabled(custom);
+        if (loadingRequest != null) {
+            loadingRequest.cancel(true);
+        }
+        loadingRequest = loadStreams();
+    }
+
+    private Future<QuarkusModel> loadStreams() {
+        final String endpoint = getSelectedEndpointUrl();
+        if (endpoint == null || endpoint.isBlank()) {
+            return null;
+        }
+        showSpinner();
+        return ApplicationManager.getApplication().executeOnPooledThread(() -> {
+            try {
+                return QuarkusModelRegistry.INSTANCE.loadStreams(endpoint, new EmptyProgressIndicator());
+            } catch (Exception e) {
+                ApplicationManager.getApplication().invokeLater(() -> {
```

Comment:
`var state = ModalityState.stateForComponent(getComponent();`

and use this variable

Reply:
done